### PR TITLE
Add parameter groups with differential learning rate to the UNet

### PR DESCRIPTION
**Pull request by asbjrnmunk**
When fine tuning I want to apply a different learning rate to each block of convolution layers. Specifically, each group is given a "position" in the forward pass $k$, such that `in_conv` has "position" $k = 0$ and `out_conv` (the last convolution block) has position $k = 9$. I want to multiply the base learning rate of each block by a constant $\gamma \in (0, 1)$ such block $k$ has $\text{lr}_{k} = \text{lr}_b \cdot \gamma^k$. 

This PR implements this by calling $\gamma$ `layer_wise_lr_decay_factor` while manually grouping each parameter into their own parameter groups inside the `UNet.get_parameter_groups` method.

Further, optional patch size and batch size arguments are added to the manager and both training and finetuning.

### CODE REVIEW

**Review by Sllambias — CHANGES_REQUESTED** [review]
Different setup with a param groups method in networks and then a function outside networks to set LR's per layer according to the function supplied.

Also we need to consider how we want to set groups. Could e.g. have a standard method that returns the names of the "named_children" defined in YuccaNet which can then be overwritten for networks where that method is not good.

**Comment by asbjrnmunk** [comment]
Closing, since this will now be part of Agave. (Where it will be appreciated more than here it seems!)
